### DISCUSSION TITLE
High Availability Controller

### DISCUSSION THREAD

**Discussion by xunholy** [Koenkk/zigbee2mqtt#6881] [Devices]
I'm currently running z2mqtt in a Kubernetes cluster with 6x RPis and 1 node has the ZZH controller connected, however this couples my implementation to that node and means I don't have the resiliency I'd ideally like if my node was no longer online.

Is it possible to somehow have multiple devices that could be used either as an active-active or failover when there is an outage at a single node layer?

Would love to hear if anyone has either attempted this or if it's just not possible today, in which I'd like to understand more about why that is the case.

**Comment by akinomeroglu** [comment]
Hi @xUnholy,

If you check the FAQ [1], it says that if you are switching between same hardware you don't need to repair your devices. As far as I know there can be only one coordinator at the same time in network so it means that you can't have an active-active infra because of Zigbee limitation. 

It seems that it is possible to implement a method for failover method. First you need two same coordinator hardware. ZZH in your case. Second, you need to mount your data directory to your failover node when something happens to your active node. This could easily achieved if you have a minio or something similar storage solution in your setup. 

Third and most difficult thing is fencing your active pi node. You must cut the active pi or at least active zzh energy before migrate your node to failover node. This is easy when you have a server with ILO or DAC support but in the case of rpi you need a smart plug or etc for cut-off. 

This is my initial thoughts on this issue. Regards,

[1]: https://www.zigbee2mqtt.io/information/FAQ.html#what-does-and-does-not-require-repairing-of-all-devices this far
